packages/protocol/contracts/common/FeeHandler.sol

```diff
@@ -0,0 +1,549 @@
+    * @param token Address of the token to set.
+    * @param newMax New sllipage to set, as Fixidity fraction.
+    */
+  function setMaxSplippage(address token, uint256 newMax) external onlyOwner {
+    _setMaxSplippage(token, newMax);
+  }
+
+  function _setMaxSplippage(address token, uint256 newMax) private {
+    TokenState storage tokenState = tokenStates[token];
+    tokenState.maxSlippage = FixidityLib.wrap(newMax);
+    emit MaxSlippageSet(token, newMax);
+  }
+
+  /**
+    * @notice Allows owner to set the daily burn limit for a token.
+    * @param token Address of the token to set.
+    * @param newLimit The new limit to set, in the token units.
+    */
+  function setDailySellLimit(address token, uint256 newLimit) external onlyOwner {
+    _setDailySellLimit(token, newLimit);
+  }
+
+  function _setDailySellLimit(address token, uint256 newLimit) private {
+    TokenState storage tokenState = tokenStates[token];
+    tokenState.dailyBurnLimit = newLimit;
+    emit DailyLimitSet(token, newLimit);
+  }
+
+  /**
+  @dev Burns CELO tokens according to burnFraction.
+  */
+  function burnCelo() external {
+    return _burnCelo();
+  }
+
+  /**
+    @dev Distributes the available tokens for all registered tokens to the feeBeneficiary.
+  */
+  function distributeAll() external {
+    return _distributeAll();
+  }
+
+  function _distributeAll() private {
+    for (uint256 i = 0; i < EnumerableSet.length(activeTokens); i++) {
+      address token = activeTokens.get(i);
+      _distribute(token);
+    }
+    // distribute Celo
+    _distribute(registry.getAddressForOrDie(GOLD_TOKEN_REGISTRY_ID));
+  }
+
+  /**
+    @dev Distributes the available tokens for all registered tokens to the feeBeneficiary.
+  */
+  function handleAll() external {
+    return _handleAll();
+  }
+
+  function _handleAll() private {
+    for (uint256 i = 0; i < EnumerableSet.length(activeTokens); i++) {
+      // calling _handle would trigger may burn Celo and distributions
+      // that can be just batched at the end
+      address token = activeTokens.get(i);
+      _sell(token);
+    }
+    _distributeAll(); // distributes Celo as well
+    _burnCelo();
+  }
+
+  function handle(address tokenAddress) external {
+    return _handle(tokenAddress);
+  }
+
+  function _handle(address tokenAddress) private {
+    // Celo doesn't have to be exchanged for anything
+    if (tokenAddress != registry.getAddressForOrDie(GOLD_TOKEN_REGISTRY_ID)) {
+      _sell(tokenAddress);
+    }
+    _burnCelo();
+    _distribute(tokenAddress);
+    _distribute(registry.getAddressForOrDie(GOLD_TOKEN_REGISTRY_ID));
+  }
+
+  /**
+    * @notice Burns all the Celo balance of this contract.
+    */
+  function _burnCelo() private {
+    TokenState storage tokenState = tokenStates[registry.getAddressForOrDie(
+      GOLD_TOKEN_REGISTRY_ID
+    )];
+    ICeloToken celo = ICeloToken(registry.getAddressForOrDie(GOLD_TOKEN_REGISTRY_ID));
+
+    uint256 balanceOfCelo = celo.balanceOf(address(this));
```

Comment:
I believe  address(this).getTotalCelo() is bit more gas friendly

Reply:
I guess you mean `address(this).balance`?